### PR TITLE
deseq2/differential to nf-test

### PR DESCRIPTION
**Pull request by pinin4fjords**
In this PR:

- Pytest to nf-test for deseq2/differential
- Add missing stub
- Correct meta.yml
- Simplify rounding to reduce precision in the R template (used to make snapshots consistent)

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/deseq2/differential/tests/main.nf.test [hunk 1]
```diff
@@ -157,7 +145,10 @@ nextflow_process {
                     file(rnaseq_test_data_dir + 'SRP254919.samplesheet.csv', checkIfExists: true),
                     file(rnaseq_test_data_dir + 'SRP254919.salmon.merged.gene_counts.top1000cov.tsv', checkIfExists: true)
                 ]
-                ch_spikes = SPOOF_SPIKES.out.map{tuple(['id':'ERCC'], it)}
+                ch_spikes = [
+                    ['id':'ERCC'], 
+                    file(rnaseq_test_data_dir + 'SRP254919.spikes.tsv', checkIfExists: true)
+                ]
```
Comment (mashehu) [inline modules/nf-core/deseq2/differential/tests/main.nf.test:151]: I don't think rnaseq_test_data_dir is the correct path in the modules repo?

Comment (pinin4fjords) [reply modules/nf-core/deseq2/differential/tests/main.nf.test:151]: It's just a convenience thing (read up) - I just didn't want to keep writing `params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/`

Comment (mashehu) [reply modules/nf-core/deseq2/differential/tests/main.nf.test:151]: yep, still not a correct variable name for a module. They should be pipeline independent.

Comment (mashehu) [reply modules/nf-core/deseq2/differential/tests/main.nf.test:151]: so `expression_test_data_dir` looks better to me 🙂 (otherwise people might be confused that it is using the test data for the rnaseq pipeline)

Comment (pinin4fjords) [reply modules/nf-core/deseq2/differential/tests/main.nf.test:151]: Oh right, gotcha now, will do.

File: modules/nf-core/deseq2/differential/contrasts_matrix_lengths.config
Comment (maxulysse) [inline modules/nf-core/deseq2/differential/contrasts_matrix_lengths.config:1]: why is this file here?

**Review by maxulysse — APPROVED** [review]
minor question

File: modules/nf-core/deseq2/differential/contrasts_matrix_lengths.config
Comment (pinin4fjords) [reply modules/nf-core/deseq2/differential/contrasts_matrix_lengths.config:1]: Because I'm an idiot - thanks ;-)
